### PR TITLE
Symantec DLP incedents fetch and update API

### PR DESCRIPTION
**Pull request by ssudheerk99**
## Status
Ready

## Description
This integration can be used to fetch and update incidents of Symantec DLP

## Related PRs
List related PRs against other branches:

## Required version of Demisto
4.0 

## Does it break backward compatibility?
   - Yes
       - Further details:
   - No

## Note:-
As Symantec DLP supports SOAP API the integration uses zeep python package. which is available in dockerhub as ssudheerk99/zeep

### CODE REVIEW

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+            self.basic = HTTPBasicAuth(username, password)
+            self.host = host
+
+        def __call__(self, r):
+            if r.url.startswith(self.host):
+                return self.basic(r)
+            else:
+                return r
+
+    serverIp=demisto.params()['server-ip']
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: For all the global params
`serverIp` => `SERVER_IP`
`username` => `USERNAME`
...

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    wsdl = 'https://{}/ProtectManager/services/v2011/incidents?wsdl'.format(serverIp)
+
+    session = Session()
+    session.auth = SymantecAuth(username,password, "https://{}".format(serverIp))
+    session.verify = False
+    transport = Transport(session=session)
+
+    client = Client(wsdl=wsdl, transport=transport)
+
+    def  getIncidentId(dtobj):
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: for all functions, use pep8 conventions 
`getIncidentId` => `get_incident_id`

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+        if type(dtobj) is datetime.datetime:
+            return client.service.incidentList(savedReportId=savedReportId, incidentCreationDateLaterThan=dtobj)
+        else:
+            raise ValueError('excpected datetime object')
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: for more nicer error message and not stack trace, use `return_error('some error here')`

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    def myconverter(o):
+        if isinstance(o, datetime.datetime):
+            return o.__str__()
+
+
+
+    if demisto.command() == 'fetch-incidents':
+        fetchincidents()
+    elif demisto.command() =='test-module':
+        demisto.results("ok")
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: Add meaningful test here. Currently the test always will be successful

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    elif demisto.command() == 'DLP-incedent-details':
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]:
```suggestion
    elif demisto.command() == 'dlp-incedent-details':
```

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+  name: isFetch
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: Incident type
+  name: incidentType
+  defaultvalue: ""
+  type: 13
+  required: false
+- display: saved report Id
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]:
```suggestion
- display: Saved Report ID
```

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+description: Integration to fetch and update symantec DLP Incidents
+detaileddescription: Symantec DLP integration uses SOAP API to communicate with Enforce
+  server to fetch and update  incident Data. uses Zeep package for SOAP Api
+configuration:
+- display: 'credentials '
+  name: credentials
+  defaultvalue: ""
+  type: 9
+  required: false
+- display: server-ip
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: display of params should be nice text with capital case (e.g Server IP)
Is it ip here or `Server URL`?

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 8]
```diff
@@ -0,0 +1,178 @@
+commonfields:
+  id: symantec DLP fetch incidents and update
+  version: -1
+name: symantec DLP fetch incidents and update
+display: symantec DLP
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]:
```suggestion
display: Symantec DLP
```

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 9]
```diff
@@ -0,0 +1,178 @@
+commonfields:
+  id: symantec DLP fetch incidents and update
+  version: -1
+name: symantec DLP fetch incidents and update
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]:
```suggestion
name: Symantec DLP
```

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 10]
```diff
@@ -0,0 +1,178 @@
+commonfields:
+  id: symantec DLP fetch incidents and update
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]:
```suggestion
  id: Symantec DLP
```

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml (hunk 3 above)
Comment (ssudheerk99) [reply Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: handled the exception with return_error()

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml (hunk 4 above)
Comment (ssudheerk99) [reply Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: added a test to pull a list of incidents

**Comment by ssudheerk99** [comment]
Hi @anara123,
made the requested changes, let me know if any thing else needs to be changed

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+    def incidentDetail(_id):
+
+        return client.service.incidentDetail(incidentId=_id,includeViolations=True,includeHistory=True)
+
+
+
+    def fetchincidents():
+        last_run=demisto.getLastRun()
+        if 'lt_time' not in last_run:
+            last_run['lt_time'] = datetime.datetime.strptime("01/11/2018 00:00:00.000000",'%d/%m/%Y %H:%M:%S.%f')
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: "01/11/2018 00:00:00.000000" is hardcoded date.
For the first fetch we usually add integration parameter `Fetch since` and the value could `2 days`, `1 month`, etc. You can use as an example https://github.com/demisto/content/blob/master/Integrations/integration-OTRS.yml#L59

Comment (ssudheerk99) [reply Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: @anara123  one doubt does "parse_date_range( )" already implemented in Demisto or do I need to explicitly implement it cause in the example integration you mentioned, I could not find it implemented anywhere
if it is already implemented can I get a reference for that

Comment (Itay4) [reply Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: @ssudheerk99 It is already implemented in our `CommonServerPython` script, which you can find here https://github.com/demisto/content/blob/master/Scripts/script-CommonServerPython.yml

Comment (ssudheerk99) [reply Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: @Itay4 thanks for the reference I have implemented the method before I saw you I comment I let that it be as both seems to be same :-p .
@anara123  I have added the fetch time parameter

let me know if any more changes to be made :-)

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+    elif demisto.command() == 'dlp-incedent-details':
+        inc_details = incidentDetail( demisto.args()['incidentId'])
+        res = helpers.serialize_object(inc_details[0])
+        demisto.results(json.loads(json.dumps(res,default = myconverter)))
+    elif demisto.command() == 'dlp-update-incident':
+        demisto.results(updateIncidents())
+
+  type: python
+  commands:
+  - name: dlp-incedent-details
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: it seems that `dlp-incedent-details` returns incident details. So you need to add outputs to the command.
https://github.com/demisto/content/blob/master/docs/context_and_ouputs/README.MD here you have READE on outputs.
Outputs are important so that in the playbook, task B will be able to use the outputs of task A

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+            for incident1 in incidentids['incidentId']:
+                inc=incidentDetail(str(incident1))
+                incidents.append({
+                    'name':"DLP Incident #"+str(incident1),
+                    'details':  inc[0]['incident']['violatedPolicyRule'][0]['ruleName']+ "RULE has been violated"
+                })
+                if inc[0]['incident']['incidentCreationDate'].isoformat() > last_run['lt_time'].isoformat():
+                    demisto.setLastRun({'lt_time':inc[0]['incident']['incidentCreationDate'].strftime("%d/%m/%Y %H:%M:%S.%f")})
+        demisto.incidents(incidents)
+        return(incidents)
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: no need to return incidents, you can remove it

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+            last_run['lt_time']= datetime.datetime.strptime(last_run['lt_time'],'%d/%m/%Y %H:%M:%S.%f')
+
+        incidentids=get_incident_id(last_run['lt_time'])
+        incidents=[]
+        if len(incidentids['incidentId']) > 0:
+            for incident1 in incidentids['incidentId']:
+                inc=incidentDetail(str(incident1))
+                incidents.append({
+                    'name':"DLP Incident #"+str(incident1),
+                    'details':  inc[0]['incident']['violatedPolicyRule'][0]['ruleName']+ "RULE has been violated"
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: you should add here another property called `"rawJSON": json.dumps(inc[0]['incident])`
You want to allow users to map fields of incidents from DLP to incident in Demisto (that is why we need the rawJSON).
rawJSON should contain JSON string of original incident/case/alert/event from the integration

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+      predefined:
+      - New
+      - Escalated
+      - Investigation
+      - Resolved
+      - Dissmissed
+      description: 'changes the status of the incident in DLP console NOTE:  status
+        attributes are defined in the DLP console. '
+    description: updates Symantec DLP incidents
+  dockerimage: ssudheerk99/zeep
```
Comment (glicht) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: @ssudheerk99 for the docker image we need to manage the docker creation in our sister project: https://github.com/demisto/dockerfiles . Please see if you can open a pr there to add the proper docker files and python dependencies. If you need any help with the docker creation we will be happy to assist (you can open an issue on the dockerfiles project if you have questions).

Comment (ssudheerk99) [reply Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: @glicht  created a pr for zeep python package at https://github.com/demisto/dockerfiles 
@anara123   will update the the docker image name in the integration once it gets published in doceker hub 

:-)

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml (hunk 12 above)
Comment (ssudheerk99) [reply Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: added the output to symantec.DLP.Incident

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml (hunk 14 above)
Comment (ssudheerk99) [reply Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: added the rawJSON, also added the severity field

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml (hunk 15 above)
Comment (ssudheerk99) [reply Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: @glicht as the pr got merged when will be the zeep docker image will be available in docker hub ?

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 16]
```diff
@@ -0,0 +1,221 @@
...
+
+  type: python
+  commands:
+  - name: dlp-incident-details
+    arguments:
+    - name: incidentId
+      required: true
+      description: incident Id  from DLP console to fetch the data of
+      outputs:
+    - contextPath: Symantec.DLP.Incident
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: Please change it to be `SymantecDLP.Incident` (to make it a bit shorter) and mention the fields that incident can have. For example:
If incident have Name and ID and CreateTime field, then add context path like:
```
SymantecDLP.Incident.ID
SymantecDLP.Incident.Name
SymantecDLP.Incident.CreatedTime
```

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 17]
```diff
@@ -0,0 +1,221 @@
...
+      - INFO
+      description: changes the severity of the Incident
+    - name: status
+      auto: PREDEFINED
+      predefined:
+      - New
+      - Escalated
+      - Investigation
+      - Resolved
+      - Dissmissed
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: @ssudheerk99 changing the status to Resolved or Dissmissed will close the incident in DLP?

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 18]
```diff
@@ -0,0 +1,221 @@
...
+        'Contents':json_res,
+        'EntryContext':{'Symantec':{"DLP":{'Incident':json_res}}}
+        })
+        
+    elif demisto.command() == 'dlp-update-incident':
+        demisto.results(updateIncidents())
+
+  type: python
+  commands:
+  - name: dlp-incident-details
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: is there an API to search for an incident via filters/query? And not only via incident id?

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 19]
```diff
@@ -0,0 +1,221 @@
...
+    def incidentDetail(_id):
+
+        return client.service.incidentDetail(incidentId=_id,includeViolations=True,includeHistory=True)
+        
+    def myconverter(o):
+        if isinstance(o, datetime.datetime):
+            return o.__str__()        
+    
+        
+    def parse_date_range(daterange):
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: you didn't have to implement it. You could just use our built in command `parse_date_range`

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 20]
```diff
@@ -0,0 +1,221 @@
...
+        demisto.results(updateIncidents())
+
+  type: python
+  commands:
+  - name: dlp-incident-details
+    arguments:
+    - name: incidentId
+      required: true
+      description: incident Id  from DLP console to fetch the data of
+      outputs:
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: you have 1 extra tab / 2 spaces here.
move `outputs` back 2 spaces to align at the same level as arguments

**Comment by anara123** [comment]
@ssudheerk99 Appreciate it a lot your contribution and your fast replies/fixes. 
Can I ask you to add few (2-3) more commands to the integration, it will help us a lot. 
If yes, I will be very happy to assist and we can do a zoom session to review it together and make the process faster.

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml (hunk 17 above)
Comment (ssudheerk99) [reply Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: yes, but both are used in different ways
Dismissed is used when an alert is received and no action is to be taken and can be closed directly.
Resolved is used when an alert is received and some action has to be taken like escalating to manager or some thing else and the incident will be marked as resolved when all the process is completed.

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 21]
```diff
@@ -0,0 +1,220 @@
...
+    session.auth = SymantecAuth(USERNAME,PASSWORD, "https://{}".format(SERVER_IP))
+    session.verify = False
+    transport = Transport(session=session)
+
+    client = Client(wsdl=wsdl, transport=transport)
+
+    def  get_incident_id(dtobj):
+
+        if type(dtobj) is datetime.datetime:
+            return client.service.incidentList(savedReportId=SAVED_REPORT_ID, incidentCreationDateLaterThan=dtobj)
```
Comment (anara123) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: why `SAVED_REPORT_ID` we get this variable from params and not argument?

**Comment by glicht** [comment]
I've added a commit with the following changes:

* cache support. Also important for envs without outside internet access. Docker image has been updated with a cache to support this (See: https://github.com/demisto/dockerfiles/pull/22 )
* python typo fixes
* server host description
* changed base command name from dlp to sdlp

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml (hunk 21 above)
Comment (ssudheerk99) [reply Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: it is required to fetch incidents, that is why I preferred using it in params

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 22]
```diff
@@ -0,0 +1,336 @@
...
+description: Integration to fetch and update symantec DLP Incidents
+detaileddescription: Symantec DLP integration uses SOAP API to communicate with Enforce
+  server to fetch and update  incident Data. uses Zeep package for SOAP Api
+configuration:
+- display: 'credentials '
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: Please change to `Username`

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 23]
```diff
@@ -0,0 +1,336 @@
...
+  name: credentials
+  defaultvalue: ""
+  type: 9
+  required: false
+- display: 'Server Host (ip or hostname. Specify port followed by ":". ie: dlp-host:8443)'
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: Please change this to be the first parameter
`ip` -> `IP address`

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 24]
```diff
@@ -0,0 +1,336 @@
...
+description: Integration to fetch and update symantec DLP Incidents
+detaileddescription: Symantec DLP integration uses SOAP API to communicate with Enforce
+  server to fetch and update  incident Data. uses Zeep package for SOAP Api
+configuration:
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: Please add proxy and unsecure parameters

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 25]
```diff
@@ -0,0 +1,336 @@
...
+    from zeep.cache import SqliteCache
+    import random
+    from datetime import datetime
+    import json
+    import requests
+    from requests.packages.urllib3.exceptions import InsecureRequestWarning
+
+    requests.packages.urllib3.disable_warnings()
+
+    # hexdump:732073756468656572206b756d6172
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: Why is this comment here?

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 26]
```diff
@@ -0,0 +1,336 @@
...
+    from zeep import Client
+    from zeep.transports import Transport
+    from requests.auth import AuthBase, HTTPBasicAuth
+    from zeep import helpers
+    from zeep.cache import SqliteCache
+    import random
+    from datetime import datetime
+    import json
+    import requests
+    from requests.packages.urllib3.exceptions import InsecureRequestWarning
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: This import is not used

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 27]
```diff
@@ -0,0 +1,336 @@
...
+                return self.basic(r)
+            else:
+                return r
+
+    SERVER_IP = demisto.params()['server-ip']
+    USERNAME = demisto.params()['credentials']['identifier']
+    PASSWORD = demisto.params()['credentials']['password']
+    SAVED_REPORT_ID = demisto.params()['saved-Report-id']
+    FETCH_TIME = demisto.params()['fetch_time']
+    SEVERITY = {'high':3,'medium':2,'low':1,'info':5}
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: Please lint properly

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 28]
```diff
@@ -0,0 +1,336 @@
...
+
+    wsdl = 'https://{}/ProtectManager/services/v2011/incidents?wsdl'.format(SERVER_IP)
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: It's better to let the user to set the `https://` and not hard code it

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 29]
```diff
@@ -0,0 +1,336 @@
...
+    def  get_incident_id(dtobj):
+
+        if type(dtobj) is datetime:
+            return client.service.incidentList(savedReportId=SAVED_REPORT_ID, incidentCreationDateLaterThan=dtobj)
+        else:
+            return_error('excpected datetime object')
+
+
+
+    def incidentDetail(_id):
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: Please use PEP8, so this should be `incident_detail`
This applies to all function names

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 30]
```diff
@@ -0,0 +1,336 @@
...
+            for incident1 in incidentids['incidentId']:
+                inc=incidentDetail(str(incident1))
+                serialized_obj = helpers.serialize_object(inc[0])
+                incidents.append({
+                    'name':"#"+str(incident1) + " DLP "+ inc[0]['incident']['violatedPolicyRule'][0]['ruleName']+ "RULE has been violated",
+                    'details': json.dumps(serialized_obj,default = myconverter),
+                    'rawJSON': json.dumps(serialized_obj,default = myconverter),
+                    'severity' : SEVERITY[inc[0]['incident']['severity']]
+                })
+                if inc[0]['incident']['incidentCreationDate'].isoformat() > last_run['lt_time'].isoformat():
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: No reason to do this `last_run['lt_time'].isoformat()` for every incident
Initialize it once and use it then

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 31]
```diff
@@ -0,0 +1,336 @@
...
+        if 'lt_time' not in last_run:
+            fetch_time,_ = parse_date_range(FETCH_TIME)
+            last_run['lt_time'] = fetch_time
+        else:
+            last_run['lt_time']= datetime.strptime(last_run['lt_time'],'%d/%m/%Y %H:%M:%S.%f')
+
+        incidentids=get_incident_id(last_run['lt_time'])
+        incidents=[]
+        if len(incidentids['incidentId']) > 0:
+            for incident1 in incidentids['incidentId']:
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: Please provide meaningful variable better name than `incident1`

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 32]
```diff
@@ -0,0 +1,336 @@
...
+        demisto.incidents(incidents)
+
+
+    def updateIncidents():
+        args = demisto.args()
+        inc_id = args['incidentId']
+        args.pop('incidentId')
+        if args['status'] == 'New':
+            args['status'] = 'incident.status.New'
+        return client.service.updateIncidents(updateBatch={'batchId':str(random.randint(11111,99999)),'incidentId':inc_id,'incidentAttributes':args})
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: Please add comment to explain the logic

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 33]
```diff
@@ -0,0 +1,336 @@
...
+
+
+
+    if demisto.command() == 'fetch-incidents':
+        fetchincidents()
+
+    elif demisto.command() =='test-module':
+        try:
+            inc_list = get_incident_id(datetime.now())
+            demisto.results("ok")
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: `"` -> `'`

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 34]
```diff
@@ -0,0 +1,336 @@
...
+        except:
+            return_error("Invalid instance details")
+
+    elif demisto.command() == 'sdlp-incident-details':
+        inc_details = incidentDetail( demisto.args()['incidentId'])
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: In the `if-else` block there should be only function calling, so all the logic should be in these functions

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 35]
```diff
@@ -0,0 +1,336 @@
...
+        res = helpers.serialize_object(inc_details[0])
+        json_res = json.loads(json.dumps(res,default = myconverter))
+        demisto.results({'ContentsFormat': formats['json'],
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: Please add `HumanReadable` to the entry

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 36]
```diff
@@ -0,0 +1,336 @@
...
+            'Type': entryTypes['note'],
+            'Contents':json_res,
+            'EntryContext':{'SymantecDLP':{'Incident':json_res}}}
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: We should define the unique key as follows: 
```
'EntryContext': {
    'SymantecDLP.Incident(val.<UNIQUE KEY> && val.<UNIQUE KEY> === obj.<UNIQUE KEY>)': json_res
}
```

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 37]
```diff
@@ -0,0 +1,336 @@
...
+        )
+
+    elif demisto.command() == 'sdlp-update-incident':
+        demisto.results(updateIncidents())
+
+    elif demisto.command() == 'sdlp-incident-binaries':
+        args = demisto.args()
+        binaries = incidentBinaries(args)
+        if len(binaries["Component"]) > 0 :
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: This is a bit risky, as if `Component` is not in `binaries`, a KeyError Exception will be thrown
Please use `.get()`

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 38]
```diff
@@ -0,0 +1,336 @@
...
+
+    elif demisto.command() == 'sdlp-list-custom-attributes':
+        customAttributes = client.service.listCustomAttributes()
+        demisto.results({'ContentsFormat': formats['json'],
+                                    'Type': entryTypes['note'],
+                                    'Contents':customAttributes,
+                                    'EntryContext':{'customAttributes':customAttributes}
+                                    }
+                                )
+
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: Please remove blank lines

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 39]
```diff
@@ -0,0 +1,336 @@
...
+
+
+
+
+  type: python
+  commands:
+  - name: sdlp-incident-details
+    arguments:
+    - name: incidentId
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: Please use PEP8 convention, so `incidentId` -> `incident_id`

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 40]
```diff
@@ -0,0 +1,336 @@
...
+      required: true
+      description: incident Id  from DLP console to fetch the data of
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: Please remove unnecessary white spaces

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml [hunk 41]
```diff
@@ -0,0 +1,336 @@
...
+    outputs:
```
Comment (Itay4) [inline Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: Please add types to all outputs

File: TestData/EICAR.exe [hunk 42]
```diff
@@ -1 +0,0 @@
-X5O!P%@AP[4\PZX54(P^)7CC)7}$EICAR-STANDARD-ANTIVIRUS-TEST-FILE!$H+H*
```
Comment (Itay4) [inline TestData/EICAR.exe]: Why did you remove this ?

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml (hunk 25 above)
Comment (ssudheerk99) [reply Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: removed the comment

File: Integrations/integration-symantec-DLP-incident-reportintg-update.yml (hunk 26 above)
Comment (ssudheerk99) [reply Integrations/integration-symantec-DLP-incident-reportintg-update.yml]: updated the import, was using to disable the ssl warnings

File: TestData/EICAR.exe (hunk 42 above)
Comment (ssudheerk99) [reply TestData/EICAR.exe]: this was a test malware file which got cloned when I cloned the content repository. got deleted by AV  on my PC as it is a malware file

**Comment by Itay4** [comment]
@ssudheerk99 How's it going? do you need our assistance to continue with the PR?

**Comment by ssudheerk99** [comment]
hey @Itay4 , actually could not get time to work on this, I'm engaged in other tasks, will try to make some time for this.
